console/console-init/ui/src/modules/device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -272,6 +272,12 @@ const DeviceFilter: React.FunctionComponent<IDeviceFilterProps> = (
     dataList.push(dataObj);
     setLastAppliedFilter(dataList);
   };
+  const isEnabledRunFilter = () => {
+    if (JSON.stringify(filter) != JSON.stringify(setInitialFilter())) {
```

Comment:
you should use compareObject custom function for compare objects.
 if (!compareObject(filter,setInitialFilter())) {
}